exonum/src/encoding/fields.rs

```diff
@@ -258,6 +258,64 @@ impl<'a> Field<'a> for DateTime<Utc> {
     }
 }
 
+impl<'a> Field<'a> for Duration {
+    fn field_size() -> Offset {
+        (mem::size_of::<i64>() + mem::size_of::<i32>()) as Offset
+    }
+
+    unsafe fn read(buffer: &'a [u8], from: Offset, to: Offset) -> Self {
+        let secs = LittleEndian::read_i64(&buffer[from as usize..from as usize + 8]);
+        let nanos = LittleEndian::read_i32(&buffer[from as usize + 8..to as usize]);
+
+        // Assuming that buffer was checked and Duration object can be constructed.
+        Duration::seconds(secs) + Duration::nanoseconds(i64::from(nanos))
+    }
+
+    fn write(&self, buffer: &mut Vec<u8>, from: Offset, to: Offset) {
+        let secs = self.num_seconds();
+        let nanos_as_duration = *self - Duration::seconds(secs);
+        // Since we're working with only nanos, no overflow is expected here.
+        let nanos = nanos_as_duration.num_nanoseconds().unwrap() as i32;
+        LittleEndian::write_i64(
+            &mut buffer[from as usize..to as usize - mem::size_of_val(&nanos)],
```

Comment:
Nitpick: I'd suggest to unify the use of slice indexes in all methods; `mem::size_of::<T>()` seems the best option to me.

Reply:
Due to rust abi is unstable `mem::size_of::<T>()` can be different on the various rust versions.